bloom-housing/bloom
# feat: listings approval superadmin flow

# Pull Request Template

## Issue Overview

This PR addresses https://github.com/metrotranscom/doorway/issues/303


## Description

Adds the superadmin flow for listings approval. When a listing is in pending review (which can be created by logging in as a jurisdictional admin and submitting a listing), as a superadmin you should be able to request changes on a listing.

## How Can This Be Tested/Reviewed?

You have to enable the environment variable in partners to turn the feature on.

When a superadmin requests changes, a modal should pop that matches the Figma. The text field should be required. On submit, the status should change to changes requested and the changes requested field should be saved onto the listing. Now in the details state, the changes requested field should show, and should truncate if the text is long.

When a partner re-submits for approval, if a superadmin requests changes again the form field should populate with the existing requested changes as the default value.

Note that there is an additional field on the detail view w the requested-by-user confirmed w product/design that is not in the Figma.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
